tests/tasks/test_control_flow.py

```diff
@@ -162,6 +162,26 @@ def test_merge_imperative_flow():
     assert state.result[c].result == 2
 
 
+def test_merge_imperative_flow_checkpoint_false():
+    flow = Flow("test")
+
+    cond = identity.copy().bind(True, flow=flow)
+    with case(cond, True):
+        a = inc.copy().bind(1, flow=flow)
+
+    with case(cond, False):
+        b = inc.copy().bind(2, flow=flow)
+
+    c = merge(a, b, flow=flow, checkpoint=False)
+
+    state = flow.run()
+    assert state.result[cond].result is True
+    assert state.result[a].result == 2
+    assert state.result[b].is_skipped()
+    assert state.result[c].result == 2
+    assert c.checkpoint == False
```

Comment:
I am not sure we need the other tests right?

Reply:
Yeah, probably not. Just removed them, thank you!

Reply:
Thank you for the change and the PR! Will leave to @desertaxle to approve.

Reply:
Anytime, thanks for reviewing!